packages/get-platform/src/getPlatform.ts

```diff
@@ -54,64 +70,149 @@ export async function getos(): Promise<GetOSResult> {
+      'alpine',
+      (originalDistro) =>
+        ({
+          targetDistro: 'musl',
+          familyDistro: originalDistro,
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'raspbian',
+      (originalDistro) =>
+        ({
+          targetDistro: 'arm',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'nixos',
+      (originalDistro) =>
+        ({
+          targetDistro: 'nixos',
+          originalDistro,
+          familyDistro: 'nixos',
+        } as const),
+    )
+    .with(
+      'debian',
+      'ubuntu',
+      (originalDistro) =>
+        ({
+          targetDistro: 'debian',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const),
+    )
+    .with(
+      'rhel',
+      'centos',
+      'fedora',
+      (originalDistro) =>
+        ({
+          targetDistro: 'rhel',
+          familyDistro: 'rhel',
+          originalDistro,
+        } as const),
+    )
+    .otherwise((originalDistro) => {
+      if (idLike.includes('debian') || idLike.includes('ubuntu')) {
+        return {
+          targetDistro: 'debian',
+          familyDistro: 'debian',
+          originalDistro,
+        } as const
+      }
 
-  if (idLike.includes('centos') || idLike.includes('fedora') || idLike.includes('rhel') || id === 'fedora') {
-    return 'rhel'
-  }
+      if (id === 'arch' || idLike.includes('arch')) {
+        return {
+          targetDistro: 'debian',
+          familyDistro: 'arch',
+          originalDistro,
+        } as const
+      }
 
-  if (idLike.includes('debian') || idLike.includes('ubuntu') || id === 'debian') {
-    return 'debian'
-  }
+      if (idLike.includes('centos') || idLike.includes('fedora') || idLike.includes('rhel')) {
+        return {
+          targetDistro: 'rhel',
+          familyDistro: 'rhel',
+          originalDistro,
+        } as const
+      }
+
+      /* Generic distro info fallback */
+      return {
+        targetDistro: undefined,
+        familyDistro: undefined,
+        originalDistro,
+      } as const
+    })
 
-  return
+  debug(`Found distro info:\n${JSON.stringify(distroInfo, null, 2)}`)
+  return distroInfo
 }
 
-export async function resolveDistro(): Promise<undefined | GetOSResult['distro']> {
+export async function resolveDistro(): Promise<DistroInfo> {
   // https://github.com/retrohacker/getos/blob/master/os.json
+
   const osReleaseFile = '/etc/os-release'
-  const alpineReleaseFile = '/etc/alpine-release'
-
-  if (await exists(alpineReleaseFile)) {
-    return 'musl'
-  } else if (await exists(osReleaseFile)) {
-    return parseDistro(await readFile(osReleaseFile, 'utf-8'))
-  } else {
-    return
+  try {
+    const osReleaseInput = await readFile(osReleaseFile, { encoding: 'utf-8' })
+    return parseDistro(osReleaseInput)
+  } catch (_) {
+    return {
+      targetDistro: undefined,
+      familyDistro: undefined,
+      originalDistro: undefined,
+    }
   }
```

Comment:
No reason to check `/etc/alpine-release` separately, which only returns the version ID of Alpine, similarly to `/etc/lsb-release` on Centos.
Even in old Alpine version (e.g., the one returned by `node:12-alpine`), `/etc/os-release` exists, and it contains `ID=alpine`.